src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -396,7 +401,8 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
             SeparatedSyntaxList<TypeSyntax> typeArgumentsSyntax, // may be omitted in synthesized invocations
             Compilation currentCompilation,
             ConsList<Symbol> basesBeingResolved,
-            DiagnosticBag diagnostics)
+            DiagnosticBag diagnostics,
+            TupleNamesCheckKind deepCheckTupleNames)
```

Comment:
>deepCheckTupleNames [](start = 32, length = 19)

Consider including "Kind" in the name, or even simply using "tupleNamesCheckKind" as the name.